File: 746. Min Cost Climbing Stairs/main.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        arr = [cost[0], cost[1]]
        for i in range(2, len(cost)):
            arr.append(min(arr[i-1], arr[i-2]) + cost[i])
        return min(arr[-1], arr[-2])

File: 746. Min Cost Climbing Stairs/main.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function (cost) {
  let len = cost.length;

  for (let i = 2; i < len; i++) {
    if (cost[i - 1] < cost[i - 2]) {
      cost[i] = cost[i] + cost[i - 1];
    } else {
      cost[i] = cost[i] + cost[i - 2];
    }
  }

  if (cost[len - 1] < cost[len - 2]) {
    return cost[len - 1];
  } else {
    return cost[len - 2];
  }
};
